spec.bs

```diff
@@ -232,7 +232,105 @@ to conduct an auction to select an advertisement to display to the user, they ca
 {{Window/navigator}}.{{Navigator/runAdAuction()}} function, providing an auction configuration that
 tells the browser how to conduct the auction and which on-device recorded interests are allowed to
 bid in the auction for the chance to display their advertisement.
-  
+
+<h3 id="script-runners">Script Runners</h3>
+
+<xmp class="idl">
+[Exposed=Window, Global=Window]
+interface InterestGroupBiddingScriptRunnerGlobalScope{
+};
+
+[Exposed=Window, Global=Window]
+interface InterestGroupScoringScriptRunnerGlobalScope{
+};
+
+[Exposed=Window, Global=Window]
+interface InterestGroupReportingScriptRunnerGlobalScope{
+};
+</xmp>
+
+Issue: where do these interfaces above get exposed to?
+
+TODO: modelingSignals
+
+<xmp class="idl">
+[Exposed=InterestGroupBiddingScriptRunnerGlobalScope]
+dictionary AdSize {
+  required double width;
+  required DOMString widthUnits;
+  required double height;
+  required DOMString heightUnits;
+};
+
+[Exposed=InterestGroupBiddingScriptRunnerGlobalScope]
+dictionary AdDescriptor {
+  required DOMString render;
+  AdSize adSize;
+};
+
+[Exposed=InterestGroupBiddingScriptRunnerGlobalScope]
+dictionary GenerateBidOutput {
+  required double bid;
+  required AdDescriptor adDescriptor;
+  DOMString ad;
+  double adCost;
+  sequence<AdDescriptor> adComponentDescriptors;
+  boolean allowComponentAuction = false;
+};
+
+</xmp>
+
+TODO: define an enum LengthUnit for AdSize's widthUnits and heightUnits.
+
+<xmp class="idl">
+[Exposed=(InterestGroupBiddingScriptRunnerGlobalScope, InterestGroupScoringScriptRunnerGlobalScope, 
+InterestGroupReportingScriptRunnerGlobalScope)]
+interface ForDebuggingOnly {
+  attribute DOMString? winReportUrl;
+  attribute DOMString? lossReportUrl;
+  undefined reportAdAuctionWin(DOMString url);
+  undefined reportAdAuctionLoss(DOMString url);
+  DOMString? takeWinReportUrl();
+  DOMString? takeLossReportUrl();
+};
+</xmp>
+
+TODO: reportAdAuctionWin/loss steps.
+
+partial interface InterestGroupBiddingScriptRunnerGlobalScope {
+  boolean setBid(GenerateBidOutput bid);
+};
+
+TODO: setbid method steps.
+
+<xmp class="idl">
+partial interface InterestGroupBiddingScriptRunnerGlobalScope {
+  undefined setPriority(double priority);

```

Comment:
Is there a reason this method doesn't appear right on the actual global, but is instead inside a partial block here?

Reply:
there isn't :)
Moved to the actual global.